### PR TITLE
SWARM-1794: Fix MP FT Timeout support.

### PR DESCRIPTION
**Pull request by mkouba**
Motivation
----------
Timeout does not work as expected - the execution thread is never
interrupted unless the operation is also marked as asynchronous.

Modifications
-------------
Always use THREAD isolation strategy for timeout operations. Also modify
the way the retry logic is implemented.

Result
------
Execution thread should be interrupted when timeout occurs.

### CODE REVIEW

**Comment by kenfinnigan** [comment]
retest this please

File: testsuite/microprofile-tcks/fault-tolerance/pom.xml [hunk 1]
```diff
@@ -99,6 +99,11 @@
                      <name>java.util.logging.config.file</name>
                      <value>${project.build.testOutputDirectory}/logging.properties</value>
                   </property>
+                  <!-- Uncomment to debug arquillian test ->
```
Comment (kenfinnigan) [inline testsuite/microprofile-tcks/fault-tolerance/pom.xml]: Looks like this is what's causing the pom parsing to fail. Needs `-->` instead at the end

**Comment by mkouba** [comment]
The failure on windows machine is `BulkheadSynchRetryTest` again. @antoinesd Have you discovered something?

**Review by antoinesd — APPROVED** [review]

**Comment by antoinesd** [comment]
not yet

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**